apis/r/R/SOMAArrayBase.R

```diff
@@ -0,0 +1,38 @@
+#' SOMA Array Base Class
+#'
+#' Adds SOMA-specific functionality to the [`TileDBArray`] class.
+
+SOMAArrayBase <- R6::R6Class(
+  classname = "SOMAArrayBase",
+  inherit = TileDBArray,
+
+  active = list(
+    #' @field soma_type Retrieve the SOMA object type.
+    soma_type = function(value) {
+      if (!missing(value)) {
+        stop("`soma_type` is a read-only field", call. = FALSE)
+      }
```

Comment:
Simpler?  `stopifnot("`soma_type` is a read-only field" = missing(value)` ?  Also in these quotes simple apostrophe may be nicer than backtick.